### PR TITLE
fix: view in room crashed on second modal spawn

### PR DESCRIPTION
**Pull request by pvinis**
This PR resolves [MOPLAT-329]

### Description

The issue was that the modal presentation was not happening on main thread. fixed.

also a couple of tiny things for xcode 14 to work, since i debugged it there.

### QA Test Case(s)

| Test Case | Feature | Environment | Acceptance Criteria | Setup Instructions/Link |
| --------- | :-----: | ----------: | ------------------: | ----------------------: |
|           |         |             |                     |                         |

### PR Checklist

- [ ] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[MOPLAT-329]: https://artsyproduct.atlassian.net/browse/MOPLAT-329?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by gkartalis — DISMISSED** [review]

**Comment by pvinis** [comment]
perfect

File: ios/Artsy/NativeModules/ARTNativeScreenPresenterModule.m [hunk 1]
```diff
@@ -23,6 +23,11 @@ @implementation ARTNativeScreenPresenterModule
 
 RCT_EXPORT_MODULE()
 
+- (dispatch_queue_t)methodQueue
+{
+    return dispatch_get_main_queue();
+}
+
```
[the remark this answers is not in the document]
Comment (brainbicycle) [reply ios/Artsy/NativeModules/ARTNativeScreenPresenterModule.m:30]: nice didn't know you could specify in native modules this way! https://reactnative.dev/docs/native-modules-ios#threading

File: ios/Podfile [hunk 2]
```diff
@@ -157,6 +157,14 @@ target 'Artsy' do
       end
     end
 
+    installer.pods_project.targets.each do |target|
+        target.build_configurations.each do |config|
+            config.build_settings['EXPANDED_CODE_SIGN_IDENTITY'] = ""
+            config.build_settings['CODE_SIGNING_REQUIRED'] = "NO"
+            config.build_settings['CODE_SIGNING_ALLOWED'] = "NO"
```
[the remark this answers is not in the document]
Comment (brainbicycle) [reply ios/Podfile:164]: if it is necessary for Xcode 14 to work can we add a HACKS entry with an explanation?

[the remark this answers is not in the document]
Comment (pvinis) [reply ios/Podfile:164]: makes sense. i dont think it should mess with signing, since the betas build fine. ill add something on the hacks.

**Review by brainbicycle — APPROVED** [review]

**Pull request merged**